runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -2251,7 +2251,7 @@ void gni_setup_per_comm_dom(int cdi)
   //
   // Create communication domain.
   //
-  gni_init(&cd->nih, cdi);
+  gni_init(&cd->nih, (chpl_nodeID * comm_dom_cnt_max) + cdi);
```

Comment:
Approved, but may want to change the argument name from `cdi` to `inst_id` for `gni_init` since it's no longer just the cdi. 

Just curious -- do you see any advantage to using PMI to calculate a node local uniq value instead of a global unique value? e.g. something like `(chpl_processID * numProcesses)+cdi` We certainly won't overflow the `uint32_t` that`GNI_CdmCreate` takes so the unique global should be fine, I was just wondering.

Reply:
> ... may want to change the argument name from `cdi` to `inst_id` ...

Yeah, that's a good point. Actually, what I may do is change this around so we still pass just the `cdi` into `gni_setup_per_comm_dom()`, and then uniquefy it into an `inst_id` within there.  Which brings me to ...

> ... do you see any advantage to using PMI to calculate a node local uniq value instead of a global unique value?

That notion was the source of my "overkill" comment in the PR description. The current value is job-unique, thus more unique than needed. I could make it merely node-unique, which is the minimum we need, but in order to do that I'd have to add a call to `PMI_Get_pes_in_app_on_smp()` and then look up our own rank ID in the returned vector. That seems like a lot of work in return for a an improvement in elegance but not functionality.

Reply:
Ah, I must have blew past the PR message -- sure enough you did say all that. I agree it's not worth the effort to use PMI to calculate the node local rank. I also wasn't sure how to actually use `PMI_Get_pes_in_app_on_smp`. Like I see how to call it, but then I wasn't sure how you determine which index belonged to the current process.

Reply:
Yeah, I'm not absolutely positive either, actually, because I haven't ever used that function. But my take from the man page is that you get back a vector containing rank IDs and then in order to uniquefy yourself within the node you'd have to search that for your own rank ID (from `PMI_Get_rank_in_app()` i.e. `chpl_nodeID`) and then use the corresponding vector index as your rank within the node.

Reply:
Ah, you're right that `PMI_Get_pes_in_app_on_smp` gives you a vector of global ranks. I thought it was going to give local ranks so every node would return say 0 and 1 and I didn't understand how you'd differentiate. With a quick test program I see:


<details>

```c
#include <pmi.h>
#include <stdio.h>
#include <assert.h>

int main() {
  PMI_BOOL spawned;
  assert(PMI_Init(&spawned) == PMI_SUCCESS);

  int global_rank;
  assert(PMI_Get_rank_in_app(&global_rank) == PMI_SUCCESS);

  int num_ranks_smp;
  assert(PMI_Get_numpes_in_app_on_smp(&num_ranks_smp) == PMI_SUCCESS);

  int ranks_smp[num_ranks_smp];
  assert(PMI_Get_pes_in_app_on_smp(ranks_smp, num_ranks_smp) == PMI_SUCCESS);

  char buf[1024];
  int buf_idx = 0;
  buf_idx += sprintf(buf, "global_rank=%d sees: ", global_rank);
  for (int i=0; i<num_ranks_smp; i++) {
    buf_idx += sprintf(buf+buf_idx, "ranks_smp[%d]=%d, ", i, ranks_smp[i]);
  }
  buf[buf_idx-2] = 0;
  printf("%s\n", buf);
}
```

</details>

```
srun -n4 -N2 ./a.out | sort
global_rank=0 sees: ranks_smp[0]=0, ranks_smp[1]=1
global_rank=1 sees: ranks_smp[0]=0, ranks_smp[1]=1
global_rank=2 sees: ranks_smp[0]=2, ranks_smp[1]=3
global_rank=3 sees: ranks_smp[0]=2, ranks_smp[1]=3
```

None of this matters for this PR, I was just curious since I expect we may need this information for future work with improving how we support running with multiple processes per node and I wanted to check while it was fresh on my mind.